components/ShadowLayer/src/MDCShadowLayer.m

```diff
@@ -360,6 +367,71 @@ - (void)commonLayoutSublayers {
   _shadowPathIsInvalid = NO;
 }
 
+- (void)animateCornerRadius:(CGFloat)cornerRadius
+             timingFunction:(CAMediaTimingFunction *)timingFunction
+                   duration:(NSTimeInterval)duration {
+  CGFloat currentCornerRadius = (self.cornerRadius <= 0) ? (CGFloat)0.001 : self.cornerRadius;
+  CGFloat newCornerRadius = (cornerRadius <= 0) ? (CGFloat)0.001 : cornerRadius;
+  // Create the paths
+  UIBezierPath *currentLayerPath = [UIBezierPath bezierPathWithRoundedRect:self.bounds
+                                                              cornerRadius:currentCornerRadius];
+  UIBezierPath *newLayerPath = [UIBezierPath bezierPathWithRoundedRect:self.bounds
+                                                          cornerRadius:newCornerRadius];
+
+  UIBezierPath *currentMaskPath = [self outerMaskPath];
+  [currentMaskPath appendPath:currentLayerPath];
+  [currentMaskPath setUsesEvenOddFillRule:YES];
+
+  UIBezierPath *newMaskPath = [self outerMaskPath];
+  [newMaskPath appendPath:newLayerPath];
+  [newMaskPath setUsesEvenOddFillRule:YES];
+
+  // Animate the top layers
+  NSString *shadowPathKey = @"shadowPath";
+  CABasicAnimation *topLayerAnimation = [CABasicAnimation animationWithKeyPath:shadowPathKey];
+  topLayerAnimation.fromValue = (__bridge id)currentLayerPath.CGPath;
+  topLayerAnimation.toValue = (__bridge id)newLayerPath.CGPath;
+  topLayerAnimation.duration = duration;
+  topLayerAnimation.timingFunction = timingFunction;
+  self.topShadow.shadowPath = newLayerPath.CGPath;
```

Comment:
To be safe you should commit all of these model layer values with CATransaction actions disabled.